player/lua/ytdl_hook.lua

```diff
@@ -418,23 +418,23 @@ local function formats_to_edl(json, formats, use_all_formats)
         end
 
         local tracks = {}
-        if track.vcodec and track.vcodec ~= "none" then
+        -- "none" means it is not a video
+        -- nil means it is unknown
+        if (o.force_all_formats or track.vcodec) and track.vcodec ~= "none" then
             tracks[#tracks + 1] = {
                 media_type = "video",
                 codec = map_codec_to_mpv(track.vcodec),
             }
         end
-        -- Tries to follow the strange logic that vcodec unset means it's
-        -- an audio stream, even if acodec is sometimes unset.
-        if (#tracks == 0) or (track.acodec and track.acodec ~= "none") then
+        if (o.force_all_formats or track.acodec) and track.acodec ~= "none" then
             tracks[#tracks + 1] = {
                 media_type = "audio",
                 codec = map_codec_to_mpv(track.acodec) or
                         ext_map[track.ext],
             }
         end
         if #tracks == 0 then
-            return nil
+            goto SKIP_TRACK
```

Comment:
Is there a reason to not just return nil here? I might not be following to logic correctly, but the case of having a nil track and using force all formats should be handled without this right?

Reply:
`tracks` always contains either a video track, an audio track, both or none. It gets reset for each format.
By returning nil, when there is nothing in `tracks`, no formats would get added as tracks whenever there is a format that is neither a video track, nor an audio track.